### PR TITLE
Close #801

### LINKED ISSUE #801: issue: useFieldArray root error is set to the actual field array root object, not the .root property using resolvers
### Version Number

7.34.0

### Codesandbox/Expo snack

https://codesandbox.io/s/immutable-dawn-v1iigh?file=%2Fsrc%2FApp.tsx

### Steps to reproduce

1. Go to [csb](https://codesandbox.io/s/immutable-dawn-v1iigh?file=%2Fsrc%2FApp.tsx)
2.1 See the `errors` object shape in the console
```json
{
  "fields": {
    "message": "The fields list is too small",
    "type": "too_small",
    "ref": {
      "name": "fields"
    }
  }
}
```
2.2 See the field array error message paragraph `<p>` content is empty
```jsx
{errors.fields && <p>{errors.fields.root?.message}</p>}
```

### Expected behaviour

The `errors` object shape is the same between using the **"native" `useFieldArray`** and **resolver** validation from [the `7.34.0` release](https://github.com/react-hook-form/react-hook-form/releases/tag/v7.34.0)

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bluebill1049**
- update doc on `useFieldArray` with newly added rules prop

close #801

### CODE REVIEW

**Review by Moshyfawn — CHANGES_REQUESTED** [review]

File: src/data/en/api.tsx [hunk 1]
```diff
@@ -1927,6 +1927,33 @@ setValue('notRegisteredInput', { test: '1', test2: '2' }); // ✅ sugar syntax t
                   </p>
                 </td>
               </tr>
+              <tr>
+                <td>
+                  <code>rules</code>
+                </td>
+                <td>
+                  <code className={typographyStyles.typeText}>string = id</code>
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply src/data/en/api.tsx:1935]: oops 👍

File: src/data/en/api.tsx [hunk 2]
```diff
@@ -1927,6 +1927,33 @@ setValue('notRegisteredInput', { test: '1', test2: '2' }); // ✅ sugar syntax t
...
+                <td>
+                  <p>
+                    Validation rules in the same format for{" "}
+                    <code>register</code>, which includes:
+                  </p>
+                  <p>required, minLength, maxLength, validate</p>
+                  <CodeArea
+                    url="https://codesandbox.io/s/react-hook-form-usefieldarray-rules-iyejbp?file=/src/index.js"
+                    withOutCopy
+                    rawData="rules={{ minLength: 4 }}"
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply src/data/en/api.tsx:1947]: Yes, I didn't want to include the complete example as the entire example is in the below section, but I love all the amendments.

**Review by Moshyfawn — APPROVED** [review]
Awesome, good job, Bill!

**Comment by bluebill1049** [comment]
thanks @Moshyfawn pretty much it's your update ahhaha

**Pull request merged**